lighthouse/validators/samples_declarations.py

```diff
@@ -35,48 +36,39 @@ def get_samples(request):
         else: 
             return []
 
-
+# https://stackoverflow.com/questions/46554866/efficiently-finding-duplicates-in-a-list
 def find_duplicates(sample_ids):
-    duplicate_sample_ids = []
-
-    for index in range(len(sample_ids)):
-        sample_id = sample_ids[index]
-
-        if (sample_ids.count(sample_id) > 1) and not (sample_id in duplicate_sample_ids):
-            duplicate_sample_ids.append(sample_id)
-
-    return duplicate_sample_ids
+    c = Counter(sample_ids)
+    return [k for k in c if c[k] > 1]
 
 
+# https://stackoverflow.com/questions/3462143/get-difference-between-two-lists
 def find_non_exist_samples(sample_ids):
     cursor = app.data.driver.db.samples.find(
         {"Root Sample ID": {"$in": sample_ids}}, {"Root Sample ID": 1}
     )
 
     existing_sample_ids = [val["Root Sample ID"] for val in cursor]
-    not_exist_sample_ids = [val for val in sample_ids if not (val in existing_sample_ids)]
+    return list(set(sample_ids) ^ set(existing_sample_ids))
```

Comment:
```suggestion
    return list(set(sample_ids) - set(existing_sample_ids))
```

Reply:
This was tricky, it seems that for sets symmetric difference is almost equal to normal difference so I was thinking a bit about possible cases where this could be not equal to a normal difference in this case. 
From what I've read in https://docs.python.org/2/library/sets.html ,  the operation ^ would be equivalent unless app.data.driver.db.samples.find returns elements that are not in sample_ids, which it should never happened currently in this code; however, in future if we modify the query that is performing this assumption would not be true anymore.

I would think is better to use a normal difference (-) if that is the original intention.